staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go

```diff
@@ -258,6 +258,10 @@ func (u *Unstructured) GetCreationTimestamp() metav1.Time {
 
 func (u *Unstructured) SetCreationTimestamp(timestamp metav1.Time) {
 	ts, _ := timestamp.MarshalQueryParameter()
+	if len(ts) == 0 {
+		RemoveNestedField(u.Object, "metadata", "creationTimestamp")
```

Comment:
@sttts creationtime is an `omitEmpty` field, so when it is zero value it doesn't get serialized.  While I may not detect it this way (@juanvallejo can you check if the passed `timestamp` is zero value), removing the field when this happens is reasonable.

Reply:
actually, since it's a struct, it still does (yay, go json), but logically, should not

Reply:
@deads2k 
> @juanvallejo can you check if the passed timestamp is zero value

Done